drivers/sensor/ism330dhcx/ism330dhcx.c

```diff
@@ -0,0 +1,844 @@
+		break;
+	case SENSOR_CHAN_GYRO_Y:
+		ism330dhcx_gyro_convert(val, data->gyro[1], sensitivity);
+		break;
+	case SENSOR_CHAN_GYRO_Z:
+		ism330dhcx_gyro_convert(val, data->gyro[2], sensitivity);
+		break;
+	case SENSOR_CHAN_GYRO_XYZ:
+		for (i = 0; i < 3; i++) {
+			ism330dhcx_gyro_convert(val++, data->gyro[i], sensitivity);
+		}
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int ism330dhcx_gyro_channel_get(enum sensor_channel chan,
+				       struct sensor_value *val,
+				       struct ism330dhcx_data *data)
+{
+	return ism330dhcx_gyro_get_channel(chan, val, data,
+					ISM330DHCX_DEFAULT_GYRO_SENSITIVITY);
+}
+
+#if defined(CONFIG_ISM330DHCX_ENABLE_TEMP)
+static void ism330dhcx_gyro_channel_get_temp(struct sensor_value *val,
+					  struct ism330dhcx_data *data)
+{
+	/* val = temp_sample / 256 + 25 */
+	val->val1 = data->temp_sample / 256 + 25;
+	val->val2 = (data->temp_sample % 256) * (1000000 / 256);
+}
+#endif
+
+#if defined(CONFIG_ISM330DHCX_SENSORHUB)
+static inline void ism330dhcx_magn_convert(struct sensor_value *val, int raw_val,
+					   u16_t sensitivity)
+{
+	double dval;
+
+	/* Sensitivity is exposed in mgauss/LSB */
+	dval = (double)(raw_val * sensitivity);
+	val->val1 = (s32_t)dval / 1000000;
+	val->val2 = (s32_t)dval % 1000000;
+}
+
+static inline int ism330dhcx_magn_get_channel(enum sensor_channel chan,
+					      struct sensor_value *val,
+					      struct ism330dhcx_data *data)
+{
+	s16_t sample[3];
+	int idx;
+
+	idx = ism330dhcx_shub_get_idx(SENSOR_CHAN_MAGN_XYZ);
+	if (idx < 0) {
+		LOG_DBG("external magn not supported");
+		return -ENOTSUP;
+	}
+
+
+	sample[0] = sys_le16_to_cpu((s16_t)(data->ext_data[idx][0] |
+				    (data->ext_data[idx][1] << 8)));
+	sample[1] = sys_le16_to_cpu((s16_t)(data->ext_data[idx][2] |
+				    (data->ext_data[idx][3] << 8)));
+	sample[2] = sys_le16_to_cpu((s16_t)(data->ext_data[idx][4] |
+				    (data->ext_data[idx][5] << 8)));
+
+	switch (chan) {
+	case SENSOR_CHAN_MAGN_X:
+		ism330dhcx_magn_convert(val, sample[0], data->magn_gain);
+		break;
+	case SENSOR_CHAN_MAGN_Y:
+		ism330dhcx_magn_convert(val, sample[1], data->magn_gain);
+		break;
+	case SENSOR_CHAN_MAGN_Z:
+		ism330dhcx_magn_convert(val, sample[2], data->magn_gain);
+		break;
+	case SENSOR_CHAN_MAGN_XYZ:
+		ism330dhcx_magn_convert(val, sample[0], data->magn_gain);
+		ism330dhcx_magn_convert(val + 1, sample[1], data->magn_gain);
+		ism330dhcx_magn_convert(val + 2, sample[2], data->magn_gain);
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static inline void ism330dhcx_hum_convert(struct sensor_value *val,
+					  struct ism330dhcx_data *data)
+{
+	float rh;
+	s16_t raw_val;
+	struct hts221_data *ht = &data->hts221;
+	int idx;
+
+	idx = ism330dhcx_shub_get_idx(SENSOR_CHAN_HUMIDITY);
+	if (idx < 0) {
+		LOG_DBG("external press/temp not supported");
+		return;
+	}
+
+	raw_val = sys_le16_to_cpu((s16_t)(data->ext_data[idx][0] |
+					  (data->ext_data[idx][1] << 8)));
```

Comment:
You don't need to call `sys_le16_to_cpu` here

Reply:
OK, is the compiler taking care of it?
If I remove the sys_le16_to_cpu()  it going to work on both LE and BE architectures in this way, or should I change the code?

Reply:
If I do:

axis1bit *val = data->ext_data[idx];
raw_val = sys_le16_to_cpu(*val);

Isn't it better? 
I'm saying this because I'm afraid that just removinge the sys_le16_to_cpu() would lead in something thst works for LE only.

Reply:
OK, I pushed a temptative change. Pls take a look.

Reply:
> If I remove the sys_le16_to_cpu() it going to work on both LE and BE architectures in this way

Yes. There are cases in your driver that you've used it correctly, such as in `ism330dhcx_sample_fetch_accel`. However, the cases that I marked in my previous review, you already manipulated the bytes manually with shifts and bitwise ORs into a register. The hardware, BE or LE, will write them back to memory in the right order. Doing `sys_le16_to_cpu` after the manual byte manipulation will actually break your driver on BE machines.

Reply:
> > If I remove the sys_le16_to_cpu() it going to work on both LE and BE architectures in this way
> 
> Yes. There are cases in your driver that you've used it correctly, such as in `ism330dhcx_sample_fetch_accel`. However, the cases that I marked in my previous review, you already manipulated the bytes manually with shifts and bitwise ORs into a register. The hardware, BE or LE, will write them back to memory in the right order. Doing `sys_le16_to_cpu` after the manual byte manipulation will actually break your driver on BE machines.

Yes, got it.
But I think that the manipulation I've done is ok for LE case only.
Have you checked the last code I pushed?

Reply:
> The hardware, BE or LE, will write them back to memory in the right order.

Oh, ok! I missed this point.
Got it! I'll repush with your suggested changes.